### PR TITLE
[dadguide] Split out db context, graph, and loader classes, add tests

### PR DESCRIPTION
**Pull request by RheingoldRiver**
**DO NOT MERGE THIS**

omg i wrote code sort of lol jk i just copy pasted existing code

tests of just database_context.py are (mostly) passing
   - is_rem/is_pem I'm not sure, are these based on current JP rotation?
   - I fixed get_monster_evo_gem
   - get_evolution_tree_ids does not include the base ID
   - I didn't test get_all_events or get_dungeon_by_id

however absolutely definitely this won't actually run since MonsterIndex
requires stuff that's no longer in the database manager class. I briefly
tried to make that work by further splitting up the files but that
didn't go well so I just gave up so here's what I did so far hf

### CODE REVIEW

**Comment by RheingoldRiver** [comment]
Tests:

```python
from dadguide.database_manager import DadguideDatabase
from dadguide.database_context import SqliteDbContext

database = DadguideDatabase('S:\\Documents\\Games\\PAD\\dadguide.sqlite')
ctx = SqliteDbContext(database)

assert ctx.get_active_skill_query(15)['name_en'] == 'Inferno Breath'
assert ctx.get_leader_skill_query(1202)['name_en'] == 'Fusion Soul'
assert ctx.get_awoken_skill(15)['name_en'] == 'Enhanced Water Orbs'
# print(ctx.get_awoken_skill_ids())
assert ctx.get_monsters_by_awakenings(5)[0].name_en == 'Crystal Aurora Dragon'
assert ctx.get_drop_dungeons(4)[0]['name_en'] == 'Diagoldos Descended!'
assert ctx.monster_is_farmable(4)
assert not ctx.monster_is_farmable(5156)
print(ctx.monster_in_rem(1073))  # base pandora
print(ctx.monster_in_rem(2121))
print(ctx.monster_in_pem(2120))  # arthur
assert ctx.monster_in_mp_shop(2256)  # odindra
assert not ctx.monster_in_mp_shop(5156)  # yusuke
assert ctx.get_prev_evolution_by_monster(1074) == 1073  # evo pandora
assert len(ctx.get_next_evolutions_by_monster(1074)) == 2  # evo pandora
assert ctx.get_base_monster_by_id(1074) == 1073 # evo pandora
assert len(ctx.get_evolution_by_material(5963)) == 3  # raziel gem
assert 6352 in ctx.get_evolution_tree_ids(1074)
assert ctx.monster_id_to_no(1074) == 1074  # ????
assert ctx.get_monsters_by_series(1)[0].name_en == 'Tyrra'
assert ctx.get_monsters_by_active(1)[0].name_en == 'Tyrra'
assert ctx.get_monster_evo_gem(
    'Archangel of Knowledge, Raziel', region='en').name_en == 'Archangel of Knowledge, Raziel\'s Gem'
```

**Comment by RheingoldRiver** [comment]
Obviously don't merge this

File: dadguide/database_context.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+from .database_manager import DgDungeon
+from .database_manager import DgEncounter
+from .database_manager import DgDrop
+from .database_manager import DgEvolution
+from .database_manager import DictWithAttrAccess
+from .database_manager import DgScheduledEvent
+from .database_manager import Server
+
+
+class SqliteDbContext(object):
```
Comment (TsubakiBotPad) [inline dadguide/database_context.py:20]: just call it DbContext for now

File: dadguide/database_context.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        n_evos = deque()
+        n_evos.append(base_id)
+        while len(n_evos) > 0:
+            n_evo_id = n_evos.popleft()
+            for e in self.get_next_evolutions_by_monster(n_evo_id):
+                n_evos.append(e)
+                evolution_tree.append(e)
+        return evolution_tree
+
+    def monster_id_to_no(self, monster_id, region=Server.JP):
```
Comment (yhgoh88) [inline dadguide/database_context.py:130]: unused function, just delete

File: dadguide/database_context.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+            'en': '\'s Gem',
+            'ko': ' 의 휘석'
+        }
+        if region not in gem_suffix:
+            return None
+        gem_name = name + gem_suffix.get(region)
+        return self.get_first_monster_where(
+            lambda m: getattr(m, 'name_{}'.format(region)) == gem_name and m.leader_skill_id == 10628)
+
+    def get_na_only_monsters(self):
```
Comment (yhgoh88) [inline dadguide/database_context.py:161]: unused, delete

Comment (TsubakiBotPad) [inline dadguide/database_context.py:161]: unused, pls delete

File: dadguide/database_context.py (hunk 2 above)
Comment (TsubakiBotPad) [inline dadguide/database_context.py:130]: unused, just delete

**Comment by RheingoldRiver** [comment]
Updated tests
```python
from dadguide.database_manager import DadguideDatabase
from dadguide.database_context import DbContext
from dadguide.monster_graph import MonsterGraph

database = DadguideDatabase('S:\\Documents\\Games\\PAD\\dadguide.sqlite')
graph = MonsterGraph(database)
ctx = DbContext(database, graph)

assert ctx.get_active_skill_query(15)['name_en'] == 'Inferno Breath'
assert ctx.get_leader_skill_query(1202)['name_en'] == 'Fusion Soul'
assert ctx.get_awoken_skill(15)['name_en'] == 'Enhanced Water Orbs'
# print(ctx.get_awoken_skill_ids())
assert ctx.get_monsters_by_awakenings(5)[0].name_en == 'Crystal Aurora Dragon'
assert ctx.get_monsters_by_awakenings(5)[0].evo_from_id == 26  # check the graph exists in DgMonster
assert ctx.get_drop_dungeons(4)[0]['name_en'] == 'Diagoldos Descended!'
assert ctx.monster_is_farmable(4)
assert not ctx.monster_is_farmable(5156)
print(ctx.monster_in_rem(1073))  # base pandora
print(ctx.monster_in_rem(2121))
print(ctx.monster_in_pem(2120))  # arthur
assert ctx.monster_in_mp_shop(2256)  # odindra
assert not ctx.monster_in_mp_shop(5156)  # yusuke
assert ctx.get_prev_evolution_by_monster(1074) == 1073  # evo pandora
assert len(ctx.get_next_evolutions_by_monster(1074)) == 2  # evo pandora
assert ctx.get_base_monster_by_id(1074) == 1073 # evo pandora
assert len(ctx.get_evolution_by_material(5963)) == 3  # raziel gem
assert 6352 in ctx.get_evolution_tree_ids(1074)
assert ctx.get_monsters_by_series(1)[0].name_en == 'Tyrra'
assert ctx.get_monsters_by_active(1)[0].name_en == 'Tyrra'
assert ctx.get_monster_evo_gem(
    'Archangel of Knowledge, Raziel', region='en').name_en == 'Archangel of Knowledge, Raziel\'s Gem'
```

**Comment by RheingoldRiver** [comment]
Lines 21-25 in `database_manager.py` and 29-31 in `dadguide.py` are currently commented/have a hardcoded path to my local machine to make tests work.

It would maybe be good to not have executable code in these files that depends on red and causes all tests to immediately fail if red isn't configured? I didn't want to touch that though so I just temp commented

**Pull request merged**